docs/aws/audit/cloudformationmonitoring/rules/stack_with_iam_role.mdx

```diff
@@ -23,6 +23,222 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the CloudFormation stacks.
+
+2. Once the AWS CLI is installed and configured, you can list all the CloudFormation stacks in your account using the following command:
+
+   ```
+   aws cloudformation list-stacks --query 'StackSummaries[].StackName' --output text
+   ```
+
+   This command will return a list of all stack names.
+
+3. For each stack name returned by the previous command, you can describe the stack details using the following command:
+
+   ```
+   aws cloudformation describe-stacks --stack-name <StackName>
+   ```
+
+   Replace `<StackName>` with the name of the stack you want to inspect. This command will return a JSON object that contains the details of the stack.
+
+4. To check if a CloudFormation stack has an IAM role, you need to look for the `RoleARN` field in the JSON object returned by the previous command. If the `RoleARN` field is not present or is empty, then the stack does not have an IAM role. 
+
+   You can use the following command to extract the `RoleARN` field from the JSON object:
+
+   ```
+   aws cloudformation describe-stacks --stack-name <StackName> --query 'Stacks[].RoleARN' --output text
+   ```
+
+   If this command returns an empty string, then the stack does not have an IAM role. If it returns an ARN, then the stack has an IAM role.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' module to interact with AWS services and 'json' to handle the JSON data.
+
+```python
+import boto3
+import json
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'region_name' with your actual AWS credentials and the region where your CloudFormation stacks are located.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create a CloudFormation client using the session.
+
+```python
+cf = session.client('cloudformation')
+```
+
+4. Use the 'describe_stacks' method of the CloudFormation client to get a list of all stacks. Then, iterate over each stack and check if it has an IAM role associated with it. If a stack does not have an IAM role, print its name.
+
+```python
+response = cf.describe_stacks()
+
+for stack in response['Stacks']:
+    if 'RoleARN' not in stack:
+        print(f"Stack {stack['StackName']} does not have an IAM role associated with it.")
+```
+
+This script will print the names of all CloudFormation stacks that do not have an IAM role associated with them.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>